src/main/java/apoc/schema/Schemas.java

```diff
@@ -296,12 +335,68 @@ private Boolean constraintsExistsForRelationship(String type, List<String> prope
      *
      * @return
      */
-    private Stream<ConstraintRelationshipInfo> constraintsForRelationship() {
+    private Stream<ConstraintRelationshipInfo> constraintsForRelationship(Map<String,Object> config) {
         Schema schema = db.schema();
 
-        return StreamSupport.stream(schema.getConstraints().spliterator(), false)
-                .filter(constraintDefinition -> constraintDefinition.isConstraintType(ConstraintType.RELATIONSHIP_PROPERTY_EXISTENCE))
-                .map(this::relationshipInfoFromConstraintDefinition);
+        SchemaConfig schemaConfig = new SchemaConfig(config);
+        Collection<String> includeRelationships = schemaConfig.getRelationships();
+        Collection<String> excludeRelationships = schemaConfig.getExcludeRelationships();
+
+        try ( Statement ignore = tx.acquireStatement() ) {
+            TokenRead tokenRead = tx.tokenRead();
+            Iterable<ConstraintDefinition> constraintsIterator;
+
+            if(!includeRelationships.isEmpty()) {
+                constraintsIterator = includeRelationships.stream()
+                        .filter(type -> !excludeRelationships.contains(type) && tokenRead.relationshipType(type) != -1)
+                        .flatMap(type -> {
+                            Spliterator<ConstraintDefinition> constraintsForType = schema.getConstraints(RelationshipType.withName(type)).spliterator();
+                            return StreamSupport.stream(constraintsForType, false);
+                        })
+                        .collect(Collectors.toList());
+            } else {
+                Iterable<ConstraintDefinition> allConstraints = schema.getConstraints();
+                constraintsIterator = StreamSupport.stream(allConstraints.spliterator(),false)
+                        .filter(index -> !excludeRelationships.contains(index.getRelationshipType().name()))
+                        .collect(Collectors.toList());
+            }
+
+            Stream<ConstraintRelationshipInfo> constraintRelationshipInfoStream = StreamSupport.stream(constraintsIterator.spliterator(), false)
+                    .filter(constraintDefinition -> constraintDefinition.isConstraintType(ConstraintType.RELATIONSHIP_PROPERTY_EXISTENCE))
+                    .map(this::relationshipInfoFromConstraintDefinition);
+
+            return constraintRelationshipInfoStream;
+        }
+    }
+
+    private Stream<GraphResult> filterResultStream(Set<String> excludes, Stream<GraphResult> graphResultStream) {
+        if (excludes == null || excludes.isEmpty()) return graphResultStream;
+        return graphResultStream.map(gr -> {
+            Iterator<Node> it = gr.nodes.iterator();
```

Comment:
Cerchiamo di dare sempre nomi significativi allev variabili:
```suggestion
             Iterator<Node> nodeIterator = gr.nodes.iterator();
```